pkg/rkecerts/certs.go

```diff
@@ -181,10 +188,41 @@ func (b *Bundle) Changed() bool {
 		if item.Certificate != nil {
 			newCertPEM = string(cert.EncodeCertPEM(item.Certificate))
 		}
+
+		// kube-kubelet certificates will always be different as they are created on-demand, we need to limit replacing them only if its absolutely necessary
+		if strings.HasPrefix(item.Name, "kube-kubelet") {
+			// Check if expired
+			oldCertX509, err := cert.ParseCertsPEM(oldCertPEM)
+			if err != nil {
+				logrus.Errorf("Error parsing old certificate PEM for [%s]: %v", item.Name, err)
+			}
+			now := time.Now()
+			if len(oldCertX509) > 0 {
+				if now.After(oldCertX509[0].NotAfter) {
+					logrus.Infof("Bundle changed: now [%v] is after certificate NotAfter [%v] for certificate [%s]", now, oldCertX509[0].NotAfter, item.Name)
+					return true
+				}
+			}
+			// Check if AltNames changed
+			if newCertPEM != "" {
+				newCertX509, err := cert.ParseCertsPEM([]byte(newCertPEM))
+				if err != nil {
+					logrus.Errorf("Error parsing new certificate PEM for [%s]: %v", item.Name, err)
+				}
+				if len(newCertX509) > 0 {
+					if !reflect.DeepEqual(oldCertX509[0].DNSNames, newCertX509[0].DNSNames) || !pki.DeepEqualIPsAltNames(oldCertX509[0].IPAddresses, newCertX509[0].IPAddresses) {
```

Comment:
Do you know if DNSNames and IPAddresses slices are sorted? If they are not, we would need to sort them before calling DeepEqual operations.